crates/services/src/service.rs

```diff
@@ -228,12 +236,13 @@ where
     tokio::task::spawn(
         async move {
             tracing::debug!("running");
-            let join_handler = run(service, stop_sender.clone());
+            let run = std::panic::AssertUnwindSafe(run(service, stop_sender.clone()));
             tracing::debug!("awaiting run");
-            let result = join_handler.await;
+            let result = run.catch_unwind().await;
```

Comment:
Doesn't this panic need to be resumed?

Reply:
It is useless because we have an anonymous task. But we can do that. Why not=)

Reply:
It just means drops are called for the task